### PR TITLE
Sprint 31

### PR DESCRIPTION
**Pull request by ericsorenson**
# Sprint 31

## [1129 Docket Clerk: Edit Case Information](https://github.com/flexion/ef-cms/issues/1129)
image

## [1925 Court User: QC of Eligible Cases](https://github.com/flexion/ef-cms/issues/1925)
image

## [573 Filing Fee: Notate how and when paid or waived](https://github.com/flexion/ef-cms/issues/573)
image

## [1900 Court User: Editing Trial Session details](https://github.com/flexion/ef-cms/issues/1900)
image

## [3290 Petitions/Docket Clerks: Flatbed and 2-Sided Scanning](https://github.com/flexion/ef-cms/issues/3290)
image

## [3917 Fix document codes](https://github.com/flexion/ef-cms/issues/3917)
image

### CODE REVIEW

**Comment by waldoj** [comment]
### PR Review Checklist

- [ ] begin by deleting any existing copy of the Git repository, running `docker system prune` to remove any cached artifacts from any prior code review
- [ ] clone a fresh copy of the Git repository (`git clone https://github.com/ustaxcourt/ef-cms.git`), fetch the pull request and name it for the sprint (e.g., `git fetch origin pull/204/head:sprint-43`), and then switch to that branch (e.g. `git checkout sprint-43`)
- [ ] review the pull request itself, to get oriented
	- [ ] read the description of the pull request, which should summarize the changes made
	- [ ] read through every task on the Scrum board that's encompassed by this pull request
	- [ ] read the description of the commits that comprise the pull request
- [ ] run all tests and generate test data with `./test-all.sh`
- [ ] stand up the site locally, with `./docker-run.sh`
	- [ ] test each new story in all major browsers
		- [ ] for internal Court functionality, perform the most thorough testing in Chrome, though also test in Edge and Firefox
		- [ ] for public-facing functionality, test in browsers consistent with [public browser use data](https://analytics.usa.gov/), including Internet Explorer 11
		- [ ] test in Mobile Safari and Mobile Chrome, with the caveat that not all internal Court functionality will be necessary on these platforms
	- [ ] use an automated audit tool for code quality and practices (recommended: [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/), aka [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk))
		- [ ] look at efficiency of page loads, asset sizes, HTTP connection management, etc.
	- [ ] review for accessibility
		- [ ] use an automated audit tool, such as Chrome Audit or aXe
		- [ ] navigate site only with the keyboard
		- [ ] use VoiceOver or Narrator to navigate the site with audio only, with the display turned off
		- [ ] manually test anything that pa11y cannot test automatically (e.g., contrast of text over images)
- [ ] [review static code analysis results](https://sonarcloud.io/organizations/flexion-github/projects) in the vendor’s SonarCloud environment
- [ ] examine OWASP ZAP output in `docs/`, to ensure that any errors are known to be false positives or have been previously declared to be acceptable
- [ ] skim all new code, in the context of existing code, [looking for problems](#what-we-look-for) (knowing that the vast majority of new code will be covered by tests)
- [ ] review all tests
	- [ ] look at code coverage of tests in [the vendor's SonarCloud environment](https://sonarcloud.io/organizations/flexion-github/projects)
		- [ ] review untested code rigorously
	- [ ] methodically review all new tests for correctness, quality of naming
- [ ] review documentation to ensure that it matches changes
- [ ] provide comments on the pull request on GitHub, as necessary
	- [ ] for comments that are specific to a particular line of code, comment on those specific lines
	- [ ] for comments that are more general, attach the comment to a random line in `README.md` (as opposed to commenting on the pull request itself), to be able to use GitHub's ability to thread discussions on those comments
- [ ] for each feature-level bug (i.e., it’s working as designed, but designed wrong), open a new issue and put it in the backlog

**Comment by waldoj** [comment]
Because GitHub lacks effective pull-request-level feedback mechanisms that must be addressed before a pull request can be merged, I'll be using `README.md` as the place to attach PR-level comments.

File: README.md [hunk 1]
```diff
@@ -252,3 +252,9 @@ Follow these steps for creating the end of sprint PRs for the court.
 8. When PR comments come in, make changes to master to fix the comments
 9. After the court approves and merges PR, merge master into develop
 10. Create a release in GitHub as sprint_00x against master and put the same description planned to be in the PR description for the court
+
+## Accessibility HTML_CodeSniffer Bookmarklet
```
[the remark this answers is not in the document]
Comment (matthopson) [reply README.md]: @waldoj I don't see that this file exists

File: web-client/src/presenter/computeds/orderTypesHelper.test.js [hunk 2]
```diff
@@ -59,4 +50,27 @@ describe('orderTypesHelper', () => {
...
+
+  it('should return showDocumentTitleInput true and documentTitleInputLabel if state.form.eventCode is NOT', () => {
+    const result = runCompute(orderTypesHelper, {
+      state: { form: { eventCode: 'NOT' } },
+    });
+    expect(result.showDocumentTitleInput).toEqual(true);
+    expect(result.documentTitleInputLabel).toEqual('Notice title');
+  });
+
+  it('should return showDocumentTitleInput false if state.form.eventCode is not O or NOT', () => {
```
[the remark this answers is not in the document]
Comment (matthopson) [reply web-client/src/presenter/computeds/orderTypesHelper.test.js]: Because `OTHER` is NOT `O` or `NOT`, producing a falsy result.

File: docs/SETUP.md [hunk 3]
```diff
@@ -53,11 +54,33 @@
...
      - `POST_CONFIRMATION_ROLE_ARN_DEV` (the ARN output after running Terraform in the `iam/terraform/environment-specific/main` dir)
      - `POST_CONFIRMATION_ROLE_ARN_STG` (the ARN output after running Terraform in the `iam/terraform/environment-specific/main` dir)
+     - `POST_CONFIRMATION_ROLE_ARN_TEST` (the ARN output after running Terraform in the `iam/terraform/environment-specific/main` dir)
      - `POST_CONFIRMATION_ROLE_ARN_PROD` (the ARN output after running Terraform in the `iam/terraform/environment-specific/main` dir)
      - `SES_DMARC_EMAIL` (email address used with SES to which aggregate DMARC validations are sent)
 8. Run a build in CircleCI.
+
+## Setting up a new environment
+1. Choose a name for the branch which will be used for deployments (henceforth `$BRANCH`). Examples are 'master', 'develop', 'staging'.
+2. Choose a name for this environment (henceforth `$ENVIRONMENT`), name limited to 4 characters. Examples are 'prod', 'dev', 'stg'.
```
[the remark this answers is not in the document]
Comment (matthopson) [reply docs/SETUP.md]: I'm trying to track down more info on this. We do know that when trying to create envs with longer names it threw up - but I'm struggling to find any documentation that specifies this.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (waldoj) [reply README.md]: Sorry, that should be prefixed with `shared/`!

File: docs/SETUP.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (matthopson) [reply docs/SETUP.md]: @kkoskelin Could you share your experiences with this?

File: shared/src/business/useCases/courtIssuedOrder/fileCourtIssuedOrderInteractor.test.js [hunk 4]
```diff
@@ -32,122 +29,101 @@ describe('fileCourtIssuedOrderInteractor', () => {
...
-      updateCaseSpy.getCall(0).args[0].caseToUpdate.documents.length,
+      updateCaseSpy.mock.calls[0][0].caseToUpdate.documents.length,
     ).toEqual(4);
     expect(
-      updateCaseSpy.getCall(0).args[0].caseToUpdate.documents[3],
+      updateCaseSpy.mock.calls[0][0].caseToUpdate.documents[3],
     ).toMatchObject({ freeText: 'Order to do anything' });
   });
+
+  it('should add a generic notice document to case, set freeText to the document title, and set the document to signed', async () => {
```
[the remark this answers is not in the document]
Comment (matthopson) [reply shared/src/business/useCases/courtIssuedOrder/fileCourtIssuedOrderInteractor.test.js]: I'm adding an additional assertion to cover this. Thanks!

File: docs/SETUP.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (kkoskelin) [reply docs/SETUP.md]: I'm hoping maybe @codyseibert or @ericsorenson or @sutt0n may remember the exact reason -- or can correct the record if I'm confused in some fashion.  We once tried to make 'staging' but ended up renaming it 'stg' as I recall.

[the remark this answers is not in the document]
Comment (waldoj) [reply docs/SETUP.md]: I just know that this is gonna be a mystery months or years down the line, and somebody's gonna get frustrated about it, so best to document now what that restriction is.

File: web-client/src/presenter/computeds/orderTypesHelper.test.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (waldoj) [reply web-client/src/presenter/computeds/orderTypesHelper.test.js]: Ohh, I got lost in `not`s. `(not 0) and (not NOT)`. Got it!

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (matthopson) [reply README.md]: It must have been refactored away. 404.

File: docs/SETUP.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (matthopson) [reply docs/SETUP.md]: Yeah, after another search - I can't find docs on this. The closest thing I've found is EB environment names have a MIN of four characters. Perhaps we should just remove that part from the documentation and add it back in when we hit an issue we can properly identify. @ericsorenson thoughts?

[the remark this answers is not in the document]
Comment (matthopson) [reply docs/SETUP.md]: @waldoj I'm just going to remove it.

[the remark this answers is not in the document]
Comment (waldoj) [reply docs/SETUP.md]: OK!

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (waldoj) [reply README.md]: I am so confused, but I'll sort that out on my own. Something has clearly gone awry with my local copy of the repo.

**Comment by waldoj** [comment]
@JessicaMarine, I have reviewed this PR to `staging`, and I recommend merging it.
